File: lib_res_tools/src/main/java/com/caesarlib/res_tools/AppNormalTool.java
package com.caesarlib.res_tools;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.net.Uri;
import android.widget.Toast;


/**
 * describe:app其他的一些工具
 * author: jihan
 * date: 2017/1/26.
 */

public class AppNormalTool {
    public final static int PERMISSION_REQUEST = 24;

    /**
     * 获取app版本号
     *
     * @return 版本号
     */
    public static int getVersionCode(Context context) {
        PackageManager pm = context.getPackageManager();
        try {
            PackageInfo pi = pm.getPackageInfo(context.getPackageName(), 0);
            return pi.versionCode;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return 1;
    }

    /**
     * 跳转权限管理界面
     *
     * @param activity 页面
     */
    public static void goPermissionSettingPage(Activity activity) {
        Uri uri = Uri.parse("package:" + activity.getPackageName());//包名
        Intent intent = new Intent("android.settings.APPLICATION_DETAILS_SETTINGS", uri);
        activity.startActivityForResult(intent, PERMISSION_REQUEST);
    }

    /**
     * 判断网络是否连接
     *
     * @return 结果
     */
    public static boolean isNetworkAvailable(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        if (cm != null) {
            //如果仅仅是用来判断网络连接
            NetworkInfo[] info = cm.getAllNetworkInfo();
            if (info != null) {
                for (int i = 0; i < info.length; i++) {
                    if (info[i].getState() == NetworkInfo.State.CONNECTED) {
                        return true;
                    }
                }
            }
        }
        return false;
    }


    /**
     * 将px值转换为dip或dp值，保证尺寸大小不变
     *
     * @param pxValue （DisplayMetrics类中属性density）
     * @return
     */
    public static int px2dip(Context context, float pxValue) {
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (pxValue / scale + 0.5f);
    }

    /**
     * 将dip或dp值转换为px值，保证尺寸大小不变
     *
     * @param dipValue （DisplayMetrics类中属性density）
     * @return
     */
    public static int dip2px(Context context, float dipValue) {
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dipValue * scale + 0.5f);
    }

    /**
     * 将px值转换为sp值，保证文字大小不变
     *
     * @param pxValue （DisplayMetrics类中属性scaledDensity）
     * @return
     */
    public static int px2sp(Context context, float pxValue) {
        final float fontScale = context.getResources().getDisplayMetrics().scaledDensity;
        return (int) (pxValue / fontScale + 0.5f);
    }

    /**
     * 将sp值转换为px值，保证文字大小不变
     *
     * @param spValue （DisplayMetrics类中属性scaledDensity）
     * @return
     */
    public static int sp2px(Context context, float spValue) {
        final float fontScale = context.getResources().getDisplayMetrics().scaledDensity;
        return (int) (spValue * fontScale + 0.5f);
    }

    /**
     * 通用的吐司
     *
     * @param context 上下文
     * @param msg     内容
     */
    public static void showToast(Context context, String msg) {
        Toast.makeText(context, msg, Toast.LENGTH_SHORT).show();
    }
    /**
     * 通用的吐司
     *
     * @param msg     内容
     */
    public static void showToast(String msg) {
        showToast(ToolsGroble.getAppContext(),msg);
    }

    /**
     * 获取value中的string值
     *
     * @param context
     * @param value
     * @return
     */
    public static String getValueString(Context context, int value) {
        return context.getString(value);
    }


}

File: lib_res_tools/src/main/java/com/caesarlib/res_tools/AppNormalTool.kt
package com.caesarlib.res_tools

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.net.Uri
import android.widget.Toast


/**
 * describe:app其他的一些工具
 * author: jihan
 * date: 2017/1/26.
 */

object AppNormalTool {
    val PERMISSION_REQUEST = 24

    /**
     * 获取app版本号
     *
     * @return 版本号
     */
    fun getVersionCode(context: Context): Int {
        val pm = context.packageManager
        try {
            val pi = pm.getPackageInfo(context.packageName, 0)
            return pi.versionCode
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }

        return 1
    }

    /**
     * 跳转权限管理界面
     *
     * @param activity 页面
     */
    fun goPermissionSettingPage(activity: Activity) {
        val uri = Uri.parse("package:" + activity.packageName)//包名
        val intent = Intent("android.settings.APPLICATION_DETAILS_SETTINGS", uri)
        activity.startActivityForResult(intent, PERMISSION_REQUEST)
    }

    /**
     * 判断网络是否连接
     *
     * @return 结果
     */
    fun isNetworkAvailable(context: Context): Boolean {
        val cm = context
            .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        if (cm != null) {
            //如果仅仅是用来判断网络连接
            val info = cm.allNetworkInfo
            if (info != null) {
                for (i in info.indices) {
                    if (info[i].state == NetworkInfo.State.CONNECTED) {
                        return true
                    }
                }
            }
        }
        return false
    }


    /**
     * 将px值转换为dip或dp值，保证尺寸大小不变
     *
     * @param pxValue （DisplayMetrics类中属性density）
     * @return
     */
    fun px2dip(context: Context, pxValue: Float): Int {
        val scale = context.resources.displayMetrics.density
        return (pxValue / scale + 0.5f).toInt()
    }

    /**
     * 将dip或dp值转换为px值，保证尺寸大小不变
     *
     * @param dipValue （DisplayMetrics类中属性density）
     * @return
     */
    fun dip2px(context: Context, dipValue: Float): Int {
        val scale = context.resources.displayMetrics.density
        return (dipValue * scale + 0.5f).toInt()
    }

    /**
     * 将px值转换为sp值，保证文字大小不变
     *
     * @param pxValue （DisplayMetrics类中属性scaledDensity）
     * @return
     */
    fun px2sp(context: Context, pxValue: Float): Int {
        val fontScale = context.resources.displayMetrics.scaledDensity
        return (pxValue / fontScale + 0.5f).toInt()
    }

    /**
     * 将sp值转换为px值，保证文字大小不变
     *
     * @param spValue （DisplayMetrics类中属性scaledDensity）
     * @return
     */
    fun sp2px(context: Context, spValue: Float): Int {
        val fontScale = context.resources.displayMetrics.scaledDensity
        return (spValue * fontScale + 0.5f).toInt()
    }

    /**
     * 通用的吐司
     *
     * @param context 上下文
     * @param msg     内容
     */
    fun showToast(context: Context?, msg: String?) {
        Toast.makeText(context, msg, Toast.LENGTH_SHORT).show()
    }

    /**
     * 通用的吐司
     *
     * @param msg     内容
     */
    fun showToast(msg: String?) {
        showToast(ToolsGroble.appContext, msg)
    }

    /**
     * 获取value中的string值
     *
     * @param context
     * @param value
     * @return
     */
    fun getValueString(context: Context, value: Int): String {
        return context.getString(value)
    }


}
